llama-index-core/llama_index/core/llms/llm.py

```diff
@@ -534,6 +547,37 @@ async def astream(
 
     # -- Tool Calling --
 
+    def chat_with_tool(
+        self,
+        tools: List["BaseTool"],
+        user_msg: Optional[Union[str, ChatMessage]] = None,
+        chat_history: Optional[List[ChatMessage]] = None,
+        verbose: bool = False,
+        **kwargs: Any,
+    ) -> ChatResponse:
+        """Predict and call the tool."""
+        raise NotImplementedError("predict_tool is not supported by default.")
```

Comment:
Any reason to not implement this? This is just predict and call but a different return type ?

Reply:
Oh i see, it's lower level